src/System.Net.Http.Listener/src/System/Net/CookieExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+
+namespace System.Net
+{
+    internal static class CookieExtensions
```

Comment:
Forgot to mention this: we need access to the "variant" info, and the "ToServerString" function, on `Cookie`.  This are not publically exposed, even on the desktop CLR.  For now, I'm using reflection to get at these.